canvas.lua

```diff
@@ -116,12 +110,14 @@ function make_canvas(cache_bank, render_bank, transparency)
 			poke4(0x5f20, clipbak)
 		end,
 		pget = function(x, y)
-			local tile = tiles[(x&-8)|((y&-8)>>>16)]
+			-- local tile = tiles[(x&-8)|((y&-8)>>>16)]
+			local tile = tiles[x & -8  |  y>>>16 & 0x0.fff8]
 			if tile then
-				return (
-					ord(tile, ((y&7)<<2)|((x&7)>>>1)+1)
-					>>> ((x&1)<<2)
-				) & 15
+				return ord(tile, (y<<2 & 0x1c  |  x>>>1 & 0x03)+1)>>>(x<<2 & 0x4) & 15
+				-- return (
```

Comment:
note, there's a slight change here, with the `(x&7)>>>1` part: 
before: 7>>>1 (i.e. 0x03.8)
after: 0x03

but this shouldn't make any difference to the behavior